src/utils/segments-middleware.js

```diff
@@ -1,12 +1,14 @@
+// @flow
 import _ from "lodash";
+import { Request, Response } from "express";
 
 /**
  * @param  {Object}   req
  * @param  {Object}   res
  * @param  {Function} next
  */
 export default function segmentsMiddlewareFactory() {
-  return function segmentsMiddleware(req, res, next) {
+  return function segmentsMiddleware(req: Request, res: Response, next: Function) {
```

Comment:
We should use `NextFunction` from [express types](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/express/express-tests.ts#L20) rather than the plain `Function`.